mne/time_frequency/tfr.py

```diff
@@ -1374,3 +1376,112 @@ def tfr_multitaper(inst, freqs, n_cycles, time_bandwidth=4.0,
         out = (out, AverageTFR(info, itc, times, freqs, nave,
                                method='mutlitaper-itc'))
     return out
+
+
+def grand_average(all_tfr, drop_bads=True):
+    """Make grand average of a list TFR data
+
+    Selects the subset of good channels across all datasets.
+
+    The grand_average.nave attribute will be equal the number
+    of datasets used to calculate the grand average.
+
+    Note: Grand average TFR shall not be used for source localization.
+
+    Parameters
+    ----------
+    all_tfr : list of AverageTFR data
+        The TFR datasets.
+    drop_bads : bool
+        If True, bad MEG and EEG channels are dropped.
+
+    Returns
+    -------
+    grand_average : AverageTFR
+        The grand average data.
+
+    Notes
+    -----
+    .. versionadded:: 0.10.0
+    """
+    # check if all elements in the given list are evoked data
+    if not all(isinstance(tfr_, AverageTFR) for tfr_ in all_tfr):
+        raise ValueError("Not all the elements in list are AllTFR data")
+
+    # Copy channels to leave the original evoked datasets intact.
+    all_tfr = [tfr_.copy() for tfr_ in all_tfr]
+
+    # Interpolates if necessary
+    if drop_bads:
+        for tfr in all_tfr:
+            tfr.drop_channels(tfr.info['bads'], copy=False)
+    else:
+        pass
+        # implement good channel subset here somehow??
+
+    equalize_channels(all_tfr)  # apply equalize_channels
+    # make grand_average object using combine_tfr
+    grand_average = combine_tfr(all_tfr, weights='equal')
+    # change the grand_average.nave to the number of Evokeds
+    grand_average.nave = len(all_tfr)
+    # change comment field
+    grand_average.comment = "Grand average (n = %d)" % grand_average.nave
+    return grand_average
+
+
+def combine_tfr(all_tfr, weights='equal'):
+    """Merge AverageTFR data
+
+    Data should have the same channels and the same time instants.
+    Subtraction can be performed by passing negative weights (e.g., [1, -1]).
+
+    Parameters
+    ----------
+    all_tfr : list of AverageTFR
+        The tfr datasets.
+    weights : list of float | str
+        The weights to apply to the data of each TFR instance.
+        Can also be ``'nave'`` to weight according to tfr.nave,
+        or ``"equal"`` to use equal weighting (each weighted as ``1/N``).
+
+    Returns
+    -------
+    tfr : AverageTFR
+        The new TFR data.
+
+    Notes
+    -----
+    .. versionadded:: 0.10.0
+    """
+    tfr = all_tfr[0].copy()
+    if isinstance(weights, string_types):
```

Comment:
We want to use `string_types` as much as possible for maximum compatibilty with ASCII/Unicode strings on Python2. `isinstance(u'foo', str)` will fail on Python2, for example.